File: tests/e2e/cases/06-int-ops/min3_i32/min3_i32_verified.rs
// DEMO: Three-argument signed minimum (Rust mirror of min3_i32_verified.cpp).
//
// Covers multi-arg + signed (i32) lowering, neither of which the
// existing single-arg `add_one` Rust suite exercises.

fn min3_i32(a: i32, b: i32, c: i32) -> i32 {
    let mut m = a;
    if b < m { m = b; }
    if c < m { m = c; }
    m
}

File: tests/e2e/cases/06-int-ops/min3_i32/min3_i32_verified.cpp
/*
DEMO: Three-argument signed minimum.

Adds coverage that the existing `add_one` suite is missing:
  * three input parameters (vs one),
  * signed comparison (`<` on `int32_t` lowers to LLVM `icmp slt`).

Expected verdict: VERIFIED.
*/

#include <cstdint>

int32_t min3_i32(int32_t a, int32_t b, int32_t c) {
    int32_t m = a;
    if (b < m) m = b;
    if (c < m) m = c;
    return m;
}
